cpp/src/arrow/integration/meson.build

```diff
@@ -0,0 +1,29 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+install_headers(['json_integration.h'])
+
+exc = executable(
+    'arrow-json-integration-test',
+    sources: ['json_integration_test.cc'],
+    dependencies: [arrow_dep, rapidjson_dep, gflags_dep, gtest_dep],
```

Comment:
@kou there is some inconsistency in the tests here that you alluded to before. Most tests do not define a `main` function and therefore use `gtest_main_dep` as a dependency. However, this JSON integration test does define its own main, so you have to link just `gtest_dep` or you end up with duplicate main symbols.

I don't know the history of this test enough to know if it is intentional that some tests define their own main function while others do not. Does it make sense to have two different dependencies or should we consider removing main from this test module?

Reply:
I think that this is intentional. The test is used for an unit test and an integration test with other implementations. If the test is used for an integration test with other implementations, we pass a custom `--integration` flag. So we have a custom `main()`.

Could you use `gtest_dep` not `gtest_main_dep` here?

Reply:
Sounds good - yes this is already using gtest_dep :-)